### PR TITLE
feat: heartbeat data storage implemented with influxdb

### PR DESCRIPTION
**Pull request by appmdev**
# Description

added influxdb to store heartbeat

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x ] New feature (non-breaking change which adds functionality)

## What has been changed?

- Feature: Robot and NetApp can store heartbeat information to InfluxDB

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A: inserted multiple samples of heartbeat information for different objects as netapp and robot, successfuly visualise with influxGraph.
- [x] Test B: retreived all samples of NetApp's and Robot's as TModels via postman
- [x]  Test C: retreived all samples of NetApp's and Robot's as TModels via postman using one by one method which first get all objects ids and then quering one by one and appending to the list of TModels.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/CentralApi/Program.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@
 if (builder.Environment.IsDevelopment() == false)
     builder.Services.AddHostedService<IndexCreationService>();
 
+builder.RegisterInflux();
```
Comment (Artonus) [inline src/CentralApi/Program.cs:26]: @appmdev, you are adding the reference to influxdb in CentralApi, but not using it, and you didn't add a correct configuration in `appsettings.json` in CentralApi. Maybe you could remove it?

File: src/DataAccess/Repositories/Abstract/ITimeSeriesRepository.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿using System;
+
+namespace Middleware.DataAccess.Repositories.Abstract;
+public interface ITimeSeriesRepository
```
Comment (Artonus) [inline src/DataAccess/Repositories/Abstract/ITimeSeriesRepository.cs:4]: This interface is unused.

File: src/DataAccess/Repositories/Abstract/Influx/IInfluxRepository.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    /// <summary>
+    /// Adds a new object to the data store
+    /// </summary>
+    /// <param name="model">Model of an object to be added</param>
+    /// <returns></returns>
+    Task<TModel?> AddAsync(TModel model);
+    Task<TModel?> GetStatusByIdAsync(Guid id);
+    Task<List<TModel>> GetAllAsync();
+    Task<List<TModel>> GetAllOneByOneAsync();
+}
```
Comment (Artonus) [inline src/DataAccess/Repositories/Abstract/Influx/IInfluxRepository.cs:19]: In this file there are a few things that I would like you to correct:
1. Remove unused methods, there are four of them
2. The `TDto` is not needed in the Interface definition, only `class` uses it, leave only `TModel`.

File: src/DataAccess/Repositories/Influx/InfluxRepository.cs [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+using Middleware.Models.Domain;
+using Middleware.Models.Dto;
+using Middleware.Models.Exceptions;
+using ILogger = Serilog.ILogger;
+
+
+namespace Middleware.DataAccess.Repositories.Influx;
+public class InfluxRepository<TModel, TDto> : IInfluxRepository<TModel, TDto> where TModel : BaseModel where TDto : InfluxDto, new()
+{
+    private const string Organization = "testorg";
```
Comment (Artonus) [inline src/DataAccess/Repositories/Influx/InfluxRepository.cs:15]: change `testorg` to actual value. In the influx in AWS it is `5G-ERA`

File: src/Models/Dto/RobotStatusDto.cs [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
 
 namespace Middleware.Models.Dto;
 
 [Document(IndexName = "robotStatus-idx", StorageType = StorageType.Json, Prefixes = new[] { Prefix })]
-public class RobotStatusDto : Dto
+public class RobotStatusDto : InfluxDto
 {
     public const string Prefix = "RobotStatus";
+    // Need second opinion before merging
+    public const string Measurement = "Heartbeat";
```
Comment (Artonus) [inline src/Models/Dto/RobotStatusDto.cs:14]: I am not sure if `Measurement` field is used

File: src/DataAccess/Repositories/Abstract/ITimeSeriesRepository.cs (hunk 2 above)
Comment (appmdev) [reply src/DataAccess/Repositories/Abstract/ITimeSeriesRepository.cs]: Done

File: src/CentralApi/Program.cs (hunk 1 above)
Comment (appmdev) [reply src/CentralApi/Program.cs]: removed

File: src/DataAccess/Repositories/Abstract/Influx/IInfluxRepository.cs (hunk 3 above)
Comment (appmdev) [reply src/DataAccess/Repositories/Abstract/Influx/IInfluxRepository.cs:15]: Done

File: src/Orchestrator/Controllers/StatusController.cs [hunk 6]
```diff
@@ -151,6 +151,7 @@ public async Task<ActionResult<NetAppStatusModel>> GetNetAppStatusesAsync(bool g
         try
         {
             var status = await _heartbeatService.GetAllAppStatusesAsync(generateFakeData);
+            if (status == null) return BadRequest();
```
Comment (Artonus) [inline src/Orchestrator/Controllers/StatusController.cs:154]: @appmdev why duplicate the null check on statuses?
In the next line, we are already checking for null and returning `NotFound`. It is unnecessary to have it here.

**Review by Artonus — CHANGES_REQUESTED** [review]
If you could please remove this duplicated null check in StatusController, than we're good :)

**Review by Artonus — APPROVED** [review]

**Pull request merged**